usecase/interactor/pkmn_interactor.go

```diff
@@ -0,0 +1,35 @@
+package interactor
+
+import (
+	"github.com/jvillarreal-w/academy-go-q42021/domain/model"
+	"github.com/jvillarreal-w/academy-go-q42021/usecase/repository"
+)
+
+type pokemonInteractor struct {
+	PokemonRepository repository.PokemonRepository
+}
+
+type PokemonInteractor interface {
+	Get(p []*model.Pokemon) ([]*model.Pokemon, error)
+	GetById(p *model.Pokemon, id string) (*model.Pokemon, error)
+}
+
+func NewPokemonInteractor(r repository.PokemonRepository) PokemonInteractor {
+	return &pokemonInteractor{r}
+}
+
+func (ps *pokemonInteractor) Get(p []*model.Pokemon) ([]*model.Pokemon, error) {
+	p, err := ps.PokemonRepository.FindAll(p)
```

Comment:
If you don't need to do any more logic after this call then it's ok to just `return ps.PokemonRepository.FindAll(p)`.

Same for the function below

Reply:
Addressed in a more recent commit, thank you.